backend-api/src/functions/asyncActiveSession/handlerDependencies.ts

```diff
@@ -1,16 +1,30 @@
 import { Logger } from "../services/logging/logger";
 import { Logger as PowertoolsLogger } from "@aws-lambda-powertools/logger";
 import { MessageName, registeredLogs } from "./registeredLogs";
-import { ITokenService, TokenService } from "./tokenService/tokenService";
+import {
+  ITokenService,
+  ITokenServiceDependencies,
+  TokenService,
+} from "./tokenService/tokenService";
+import { PublicKeyGetter } from "./tokenService/publicKeyGetter";
+
+const tokenServiceDependencies: ITokenServiceDependencies = {
```

Comment:
Are we planning one expanding this? I ask because I don't immediately know what the dependencies are and have to look, which I think is fine if we have multiple dependencies, but if it's just the public key getter class, it might be easier to follow if we just call it that in the interface/dependencies. What do you think?

Reply:
I kind of agree with James on this one; however, I'd be in favour of simplifying things even more, and that means removing the TokenService. Once we've created the four services we need (i.e. `DecryptJwe`, `PublicKeyGetter`, `JwtVerifier`, `PayloadValidator` - these are just example names by the way) I don't see why we need an extra layer of complication (TokenService). The handler can very easily orchestrate the calls to the different services we will have. But that's something do be done in a separate PR (if we decide to do it).